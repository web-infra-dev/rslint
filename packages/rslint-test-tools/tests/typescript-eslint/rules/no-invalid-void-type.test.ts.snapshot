exports[`no-invalid-void-type: allowAsThisParameter: true > invalid 1`] = `
{
  "diagnostics": [],
  "errorCount": 0,
  "fileCount": 7,
  "ruleCount": 0
}
`;

exports[`no-invalid-void-type: allowInGenericTypeArguments: false > invalid 1`] = `
{
  "diagnostics": [],
  "errorCount": 0,
  "fileCount": 7,
  "ruleCount": 0
}
`;

exports[`no-invalid-void-type: allowInGenericTypeArguments: true > invalid 1`] = `
{
  "diagnostics": [],
  "errorCount": 0,
  "fileCount": 7,
  "ruleCount": 0
}
`;

exports[`no-invalid-void-type: allowInGenericTypeArguments: whitelist > invalid 1`] = `
{
  "diagnostics": [],
  "errorCount": 0,
  "fileCount": 7,
  "ruleCount": 0
}
`;
