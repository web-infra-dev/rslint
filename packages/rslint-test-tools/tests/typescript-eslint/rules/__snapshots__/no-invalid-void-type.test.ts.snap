// Rstest Snapshot v1

exports[`no-invalid-void-type > invalid 1`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "DEBUG: void parent kind is KindTypeReference",
      "messageId": "debug",
      "range": {
        "end": {
          "column": 32,
          "line": 1,
        },
        "start": {
          "column": 28,
          "line": 1,
        },
      },
      "ruleName": "no-invalid-void-type",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-invalid-void-type > invalid 2`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "DEBUG: void parent kind is KindParameter",
      "messageId": "debug",
      "range": {
        "end": {
          "column": 30,
          "line": 1,
        },
        "start": {
          "column": 26,
          "line": 1,
        },
      },
      "ruleName": "no-invalid-void-type",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-invalid-void-type > invalid 3`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "DEBUG: void parent kind is KindTypeReference",
      "messageId": "debug",
      "range": {
        "end": {
          "column": 30,
          "line": 1,
        },
        "start": {
          "column": 26,
          "line": 1,
        },
      },
      "ruleName": "no-invalid-void-type",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-invalid-void-type > invalid 4`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "DEBUG: void parent kind is KindTypeAliasDeclaration",
      "messageId": "debug",
      "range": {
        "end": {
          "column": 18,
          "line": 1,
        },
        "start": {
          "column": 14,
          "line": 1,
        },
      },
      "ruleName": "no-invalid-void-type",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;
